include/revng/Support/IRHelpers.h

```diff
@@ -1575,3 +1576,19 @@ std::unique_ptr<llvm::Module> cloneIntoContext(const llvm::Module &Module,
 /// modules being merged.
 void linkFunctionModules(std::unique_ptr<llvm::Module> &&Source,
                          std::unique_ptr<llvm::Module> &Destination);
+
+/// Helper class that allows getting the exhaustive list of functions called by
+/// a function
+class FunctionCallEnumerator {
```

Comment:
This class purpose seems to enumerate the calls, nowhere you state that you enumerate all the *transitively* reachable calls.

What about `ReachableFunctionsEnumerator`?